src/dateTimeLocal.test.jsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import TestUtils from 'react-addons-test-utils';
+import DateTimeLocalInput from './DateTimeLocalInput';
+
+
+describe('DateTimeLocal', function() {
+
+	let dateTimeComponent,
+		dateTimeInputEl;
+
+	const value = '1999-12-31T23:59:59',
+		newValue = '3000-12-31T23:59:59',
+		min = '1999-12-30T00:00:00',
+		max = '3001-12-31T23:59:59',
+		callbackSpy = jasmine.createSpy('foo');
+
+	beforeEach(() => {
+		dateTimeComponent = TestUtils.renderIntoDocument(
+			<DateTimeLocalInput
+				name='datetime'
+				value={value}
+				callback={callbackSpy}
+				min={min}
+				max={max} />);
+		dateTimeInputEl = TestUtils.findRenderedDOMComponentWithTag(dateTimeComponent, 'input');
+	});
+
+	afterEach(() => {
+		dateTimeComponent = null;
+		dateTimeInputEl = null;
+	});
+
+	it('should exist', function() {
+		expect(dateTimeInputEl).not.toBeNull();
+	});
+
+	it('should take an initial value', function() {
+		expect(dateTimeComponent.state).toEqual({ value });
+	});
+
+	it('sets state with its value', function() {
+		expect(dateTimeComponent.state).toEqual({ value });
+	});
+
+
+	it('should set a min value', function() {
+		expect(dateTimeInputEl.min).toEqual(min);
+	});
+
+	it('should set a max value', function() {
+		expect(dateTimeInputEl.max).toEqual(max);
+	});
+
+	it('should have a required attribute when specified', () => {
+		expect(dateTimeInputEl.attributes.required).not.toBeNull();
```

Comment:
`toBeNull` and `toBeUndefined` are always a little risky to use because they don't check for other falsy values - in this case, the test will _always_ pass because if `dateTimeInputEl` doesn't have a `required` attribute, `dateTimeInputEl.attributes.required` will be `undefined` (not `null`!), so the test passes.

You can use `expect(dateTimeInputEl.attributes.required).not.toBeFalsy();`